api/lcp/mirror.py

```diff
@@ -0,0 +1,151 @@
+    """
+
+    NAME = ExternalIntegration.LCP
+    SETTINGS = [
+        S3UploaderConfiguration.access_key.to_settings(),
+        S3UploaderConfiguration.secret_key.to_settings(),
+        S3UploaderConfiguration.protected_access_content_bucket.to_settings(),
+        S3UploaderConfiguration.s3_region.to_settings(),
+        S3UploaderConfiguration.s3_addressing_style.to_settings(),
+        S3UploaderConfiguration.s3_presigned_url_expiration.to_settings(),
+        S3UploaderConfiguration.url_template.to_settings(),
+        LCPMirrorConfiguration.endpoint_url.to_settings()
+    ]
+
+    def __init__(self, integration):
+        """Initializes a new instance of LCPMirror class
+
+        :param integration: External integration containing mirror's properties
+        :type integration: ExternalIntegration
+        """
+        super(LCPMirror, self).__init__(integration)
+
+        self._lcp_importer_instance = None
+
+    def _create_lcp_importer(self, collection_id):
+        """Creates a new instance of LCPImporter
+
+        :return: New instance of LCPImporter
+        :rtype: LCPImporter
+        """
+        configuration_storage = CollectionConfigurationStorage(self, collection_id)
+        configuration_factory = ConfigurationFactory()
+        hasher_factory = HasherFactory()
+        lcp_encryptor = LCPEncryptor(configuration_storage, configuration_factory)
+        lcp_server = LCPServer(configuration_storage, configuration_factory, hasher_factory)
+        lcp_importer = LCPImporter(lcp_encryptor, lcp_server)
+
+        return lcp_importer
+
+    def collection_external_integration(self, db, collection_id):
+        """Returns an external integration owned by this object
+
+        :param db: Database session
+        :type db: sqlalchemy.orm.session.Session
+
+        :return: External integration owned by this object
+        :rtype: core.model.configuration.ExternalIntegration
+        """
+        external_integration = db \
+            .query(ExternalIntegration) \
+            .join(Collection) \
+            .filter(
+                Collection.id == collection_id
+            ) \
+            .one()
+
+        return external_integration
+
+    def cover_image_root(self, bucket, data_source, scaled_size=None):
+        raise NotImplementedError()
+
+    def marc_file_root(self, bucket, library):
+        raise NotImplementedError()
+
+    def book_url(self, identifier, extension='.epub', open_access=False, data_source=None, title=None):
+        """Returns the path to the hosted EPUB file for the given identifier."""
+        bucket = self.get_bucket(
+            S3UploaderConfiguration.OA_CONTENT_BUCKET_KEY if open_access
+            else S3UploaderConfiguration.PROTECTED_CONTENT_BUCKET_KEY)
+        root = self.content_root(bucket)
+        book_url = root + self.key_join([identifier.identifier])
+
+        return book_url
```

Comment:
Is this right? This seems like code copied from the S3 integration. I don't think this type of mirror _has_ a book_url and `NotImplementedError` would be appropriate.

Reply:
`LCPMirror` has to have `book_url` implemented because it used during the import process. CM first calls `LCPMirror.book_url` and then passes its result to `LCPMirror.mirror_one`